{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "$\\gamma_i$: relative risk; $q_i$: allel frequency\n",
    "\n",
    "###Under H0: $\\gamma = 1$\n",
    "\n",
    "$q \\sim Gamma(\\rho_0, \\nu_0)$    OR     $q \\sim Gamma(\\bar{q_0}\\nu_0, \\nu_0 )$ \n",
    "\n",
    "###Under H1:\n",
    "\n",
    "$q \\sim Gamma(\\rho_1, \\nu_1)$    OR    $q_i \\sim Gamma(\\bar{q_1}v_1, v_1)$\n",
    "\n",
    "$\\gamma_i \\sim Gamma(\\bar{\\gamma}\\beta, \\beta)$\n",
    "\n",
    "###Hyperparameters\n",
    "\n",
    "$\\phi_1 = (\\bar{\\gamma}, \\beta, \\bar{q_1}, v_1)$\n",
    "\n",
    "$\\phi_0 = (\\bar{q_0}, v_0)$\n",
    "\n",
    "###The rate of a specific type of mutation:\n",
    "\n",
    "$c(\\mu_{LoF}, \\mu_{Mis3}) = c(0.074, 0.32)$  Table S3 (http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003671)\n",
    "\n",
    "###Mutation rate for each gene (in Methods)\n",
    "\n",
    "Use gene length + base content as [De novo mutations revealed by whole-exome sequencing are strongly associated with autism](http://www.nature.com/nature/journal/v485/n7397/full/nature10945.html)\n",
    "\n",
    "======\n",
    "##Improvement in Nature 13772\n",
    "\n",
    "###Two different prior distributions for $\\gamma$: de novo and inherited variants\n",
    "\n",
    "$\\gamma_d \\sim Gamma(\\bar{\\gamma_d} \\beta_d, \\beta_d) \\quad \\gamma \\sim Gamma(\\bar{\\gamma} \\beta, \\beta) $\n",
    "\n",
    "=> Change in BFs\n",
    "\n",
    "### $B = \\frac{P(x|H_1)}{P(x|H_0} = \\frac{P(x_d|H_1) \\times P(x_1, x_0 | H_1)}{P(x_d|H_0) \\times P(x_1, x_0 | H_0)} = B_d \\times B_i. $"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               [,1]     [,2]\n",
       "gamma.mean.dn  14.0    2.330\n",
       "beta.dn         4.8    2.325\n",
       "gamma.mean.CC   2.8    1.000\n",
       "beta.CC         4.0 1000.000\n",
       "rho1            0.1    0.500\n",
       "nu1           200.0  100.000\n",
       "rho0            0.1    0.500\n",
       "nu0           200.0  100.000\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "source(\"scripts/TADA/TADA.R\")\n",
    "\n",
    "#################################################################\n",
    "# Application of TADA\n",
    "#################################################################\n",
    "\n",
    "# Model parameters: two categories of mutations - LoF and mis3 mutations (\"probably damaging\" by PolyPhen2)\n",
    "mu.frac <- c(0.074, 0.32) ##Mutation rates for specific category\n",
    "#gamma.mean.dn <- c(20, 4.7)\n",
    "gamma.mean.dn <- c(14, 2.33)\n",
    "#beta.dn <- c(1,1)\n",
    "beta.dn <- c(4.8, 2.325)\n",
    "\n",
    "\n",
    "gamma.mean.CC <- c(2.8, 1.00)\n",
    "beta.CC <- c(4.0, 1000)\n",
    "rho1 <- c(0.1, 0.5)\n",
    "nu1 <- c(200, 100)\n",
    "rho0 <- c(0.1, 0.5)\n",
    "nu0 <- c(200, 100)\n",
    "hyperpar <- as.array(rbind(gamma.mean.dn, beta.dn, gamma.mean.CC, beta.CC, rho1, nu1, rho0, nu0))\n",
    "l <- 100\n",
    "#pi0 <- 0.94 # the fraction of non-risk genes\n",
    "pi0 <- 1 - 0.07419\n",
    "\n",
    "hyperpar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "    Gene mut.rate dn.LoF case.LoF ctrl.LoF trans.LoF ntrans.LoF dn.mis3\n",
       "1   A1BG 2.36e-05      0        0        1         0          0       0\n",
       "2   A1CF 2.17e-05      0        1        2         1          1       0\n",
       "3  A2LD1 6.09e-06      0        0        0         0          0       0\n",
       "4    A2M 5.64e-05      0        1        5         1          0       1\n",
       "5  A2ML1 5.70e-05      0        3       16         2          6       1\n",
       "6 A4GALT 2.42e-05      0        0        2         1          0       0\n",
       "  case.mis3 ctrl.mis3 trans.mis3 ntrans.mis3     BF.dn         BF qvalue.dn\n",
       "1         0         4          7           3 0.7641762 0.54953615 0.9308429\n",
       "2         0         7          3           7 0.7799631 0.75499580 0.9294060\n",
       "3         0         0          0           0 0.9322423 0.50824558 0.8592336\n",
       "4         4        22          6           6 0.5305202 0.29674445 0.9388012\n",
       "5         5        30          6          10 0.5270516 0.05155223 0.9388542\n",
       "6         1         5          2           1 0.7590199 0.63110457 0.9312457\n",
       "     qvalue\n",
       "1 0.9132353\n",
       "2 0.8841540\n",
       "3 0.9229287\n",
       "4 0.9456775\n",
       "5 0.9506706\n",
       "6 0.9018963\n",
       "[1] 14952\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# ASC (Autism Sequencing Consortium) data\n",
    "# The file name contains the sample size information\n",
    "# The only relevant counts are dn.LoF and dn.mis3\n",
    "#data <- read.csv(\"data/ASC_2231trios_1333trans_1601cases_5397controls.csv\", header=TRUE, as.is=TRUE)\n",
    "#ntrio <- 2231  # number of trios\n",
    "#ncase <- 1601  # number of cases\n",
    "#nctrl <- 5397  # number of controls\n",
    "#ntrans <- 1333 # number of subjects with transmission data\n",
    "#N <- list(dn=ntrio, ca=ntrans+ncase, cn=ntrans+nctrl)\n",
    "\n",
    "data <- read.csv(\"data/nature_aut_id_epi.merged.txt\", header = TRUE, sep = \" \")\n",
    "#source(\"TADA/TADA.R\")\n",
    "ntrio <- 2270 + 32 + 0 + 2508 ##AUT\n",
    "ntrio <- ntrio + 0 + 356 #add EPI\n",
    "ntrio <- ntrio + 100 + 0 + 41 + 51 ##add ID samples\n",
    "ncase <- 1601 #Only from AUT paper\n",
    "nctrl <- 5397  #Only from AUT paper\n",
    "ntrans <- 1298 ##Check this information, Only use transmitted information from AUT paper nature13772\n",
    "#ntrans <- ntrio\n",
    "\n",
    "N <- list(dn=ntrio, ca=ntrans+ncase, cn=ntrans+nctrl)\n",
    "\n",
    "\n",
    "print(head(data))\n",
    "\n",
    "sum(unlist(N))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Total counts of variants\n",
    "\n",
    "$N_d$: number of trios\n",
    "\n",
    "$N_0 = N_{cn} + N_d \\quad N_1 = N_{cs} + N_d$\n",
    "\n",
    "$X_d \\sim Pois(2\\mu\\lambda N_d) \\quad X_1 \\sim Pois(q\\lambda N_1) \\quad X_0 \\sim Pois(q N_0)$\n",
    "\n",
    "$X_0 = X_u + X_{cn}$: total number of LoF variants in the controls + the number of nontransmitted LoF variants\n",
    "\n",
    "$X_1 = X_t + X_{cs}$: total number of LoF variants in the case + the number of transmitted LoF variants\n",
    "\n",
    "###The likelihood function is the product of the three individual probabilities\n",
    "\n",
    "$x = (x_d, x_1, x_0)$: count data for a gene to be tested\n",
    "\n",
    "$P(x_d, x_1, x_0|q, \\lambda) = Pois(x_d|2\\mu\\lambda N_d).Pois(x_1|q\\lambda N_1) . Pois(x_0|qN_0)$\n",
    "\n",
    "###\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##Bayes factor for any gene:\n",
    "\n",
    "$H_1: \\gamma \\neq 1 \\quad vs. \\quad H_0: \\gamma = 1$\n",
    "\n",
    "$\\phi_1 = (\\bar{\\gamma}, \\beta, \\bar{q_1}, \\nu_1)$: the prior parameters of $H_1$\n",
    "\n",
    "$\\phi_0 = (\\bar{q_0}, \\nu_0) $: the prior parameters of $H_0$\n",
    "\n",
    "$P(x_i|H_0) = \\int p(x_i|q_i, \\gamma = 1)p(q_i|H_0)dq_i$\n",
    "\n",
    "$P(x_i|H_1 = \\int p(x_i|q_i, \\gamma) p(q_i|H_1) p(\\gamma_i |H_1) dq_i d\\gamma_i$\n",
    "\n",
    "Solve the marginal likelihood of all the data, as a function of the hyperparameters $(\\phi_1, \\phi_0)$\n",
    "\n",
    "### $P(x|\\phi_1, \\phi_0) = \\prod_{i=1}^{n}\\left[ \\pi P(x_i|H_1) + (1 - \\pi)P(x_i|H_0)\\right]$ \n",
    "\n",
    "$\\pi$: the proportion of risk genes\n",
    "\n",
    "Calculate the Bayes factor for each gene:\n",
    "\n",
    "###$B_i = \\frac{P(x_i|H_1)}{P(x_i|H_0)}$\n",
    "\n",
    "\n",
    "\n",
    "###For multiple types of mutations (assumming that the data of each type of mutation are independent of each other)\n",
    "\n",
    "$B = B_{LoF}[w.B_{Mis3} + (1 - w)]$\n",
    "\n",
    "w = 0.55"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     [,1] [,2] [,3] [,4] [,5] [,6]\n",
       "[1,]    0    0    1    0    7    7\n",
       "[2,]    0    2    3    0    3   14\n",
       "[3,]    0    0    0    0    0    0\n",
       "[4,]    0    2    5    1   10   28\n",
       "[5,]    0    5   22    1   11   40\n",
       "[6,]    0    1    2    0    3    6\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Running TADA\n",
    "counts <- as.array(cbind(data$dn.LoF, ##Xd\n",
    "                         data$case.LoF+data$trans.LoF, data$ctrl.LoF+data$ntrans.LoF, ##X1, X0\n",
    "                         data$dn.mis3, \n",
    "                         data$case.mis3+data$trans.mis3, data$ctrl.mis3+data$ntrans.mis3))\n",
    "head(counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "#Running TADA\n",
    "rs <- TADA(counts, N, data$mut.rate, mu.frac, hyperpar)\n",
    "data$BF <- rs$BF.total\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "# Estimating p-values of BFs (this is optional and slow)\n",
    "#rsp <- TADAp(counts, N, data$mut.rate, mu.frac, hyperpar, l=100)\n",
    "#data$pval.TADA <- rsp$pval\n",
    "\n",
    "##Use new method from nature13772\n",
    "\n",
    "data$pval.TADA <- data$BF*(1 - pi0)/(pi0 + (1 - pi0)*data$BF)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "# FDR estimation\n",
    "data <- data[order(-data$BF),]\n",
    "data$qvalue <- Bayesian.FDR(data$BF, pi0)$FDR\n",
    "write.csv(data, \"data/ext_TADA_results_usingNewPvalues_xem.csv\", row.names=FALSE)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         Gene mut.rate dn.LoF case.LoF ctrl.LoF trans.LoF ntrans.LoF dn.mis3\n",
       "14230   SCN2A 7.40e-05      9        2        0         0          0      12\n",
       "15819 SYNGAP1 6.55e-05      8        0        0         0          0       3\n",
       "3312     CHD8 9.25e-05      7        4        0         0          0       2\n",
       "14228   SCN1A 7.50e-05      4        1        0         0          0       7\n",
       "3306     CHD2 7.02e-05      6        1        0         0          0       1\n",
       "940    ARID1B 9.02e-05      6        0        0         0          0       1\n",
       "      case.mis3 ctrl.mis3 trans.mis3 ntrans.mis3        BF.dn           BF\n",
       "14230         4        26          5           5 6.154767e+08 5.689429e+17\n",
       "15819         3         6          4           2 2.315971e+08 6.624239e+09\n",
       "3312          7        29          4           2 6.350230e+04 1.843809e+09\n",
       "14228        11        16          9           5 9.478442e+00 7.524333e+07\n",
       "3306          7         8          5           3 1.950725e+00 1.154755e+07\n",
       "940           5        35          5           8 2.901157e+05 2.667795e+06\n",
       "         qvalue.dn       qvalue pval.TADA\n",
       "14230 0.0000000255 0.000000e+00 1.0000000\n",
       "15819 0.0000000466 9.419123e-10 1.0000000\n",
       "3312  0.0000862000 2.883942e-09 1.0000000\n",
       "14228 0.4613550760 4.362478e-08 0.9999998\n",
       "3306  0.7116799140 2.510304e-07 0.9999989\n",
       "940   0.0000180000 9.887901e-07 0.9999953\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "head(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwMEBAQF\nBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcY\nGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKior\nKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+\nPj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBR\nUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2Nk\nZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3\nd3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmK\nioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJyd\nnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+w\nsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLD\nw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW\n1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp\n6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8\n/Pz9/f3+/v7////isF19AAAZF0lEQVR4nO2dCWAURbqAu3vuSSYzk/sOOUg4cxC5whFuDHIuR3TF\nBQVPluXBLvIEdHkoCq7X0wXFi32KuwqiIuqqgMp6cKnrtQRUiBiIHJEbkgBJv+kJHcJMjqnurq7u\n6v8TAaenqn77S1dXVVdXMTxANQzpAAC8gGDKAcGUA4IpBwRTDgimHBBMOSCYckAw5YBgygHBlAOC\nKQcEUw4IphwQTDkgmHJAMOWAYMoBwZQDgikHBFMOCKYcEEw5IJhyQDDlgGDKAcGUA4IpBwRTDgim\nHBBMOSCYckAw5YBgygHBlAOCKQcEUw4IphwQTDkgmHJAMOWAYMoBwZQDgikHBFMOCKYcEEw5IJhy\nQDDlgGDKAcGUA4IpBwRTDgimHBBMOSCYckAw5cgQXPX0SoA8fzuPS/ArvyX9/wb46PMjNsHLpacF\nFGMaCKYbEEw5IJhyQDDlgGDKAcGUA4IpBwTTxrmKK/4TBNPFvMS0qQP+1eQDEEwR9b96nNmJtoq+\nxy5/BoLpYVVhhjnvFF/Y44F3L38IginhkUl3jKjtNCF5E/+c989bLn8OgqlgAcuGW7y7R3b+n8Lz\npal9z10+AoL1T9WMKM7+7ykeZ8kv5l79k9nBe5ocBMF6Z0+RPcxintjnbLy7Hb/DaU/ceMVhEKxr\n6na0s5vd053OmHs358WED+z7buA3lBF85HjwZyAYOzVvdfemum/IGpgUZeqxzGr+pJnvyBY8/BB/\noIizDK4MPACCMVM9P8ZT7Fo5KOGJlKkWG8uEf9Dct2QLZsr50slnauf8JvAACMbKoSle541X5ebl\nD41fEZ3iSrptf/PfU0Jwxm6eP+oOPACCcXJPWs+UnMeTdsa4NrgTOO/j9S19Ub7gTy+M8t3ZP8oO\nPACCsVE5Jj8+4aXH4z9OXl4UHmtLe6tFvQoI7p9i8+TxWzwrAw+AYDzULR8aN/e/P8yfMmHCNYMn\netJuO9Lq1xVoRdf+8Bm/4+Ogj0EwDi7cYrYkRMwsfKek5939Ojpc915sI4FS/eADGwI/AcHKc/re\nVHvnd3I9+UOLtntKo7P+70KbSZQSvDbs8l8L/SRNDDUtECIvRuQ7vZG126Lzb1jdK7nk9boQ0uAb\nyfqvUulpgWBOfrU1f8WaIdHZ91V3dUX0v+lMaMmUEFx/srkfJRCsKIuvntux4/ZdQ3Lbpd7oin69\n9VfKmiBb8Nkl7a2MKWtRTeABEKwcP2xeM5Xnn+4wk59ZYAvjela0nUREtuApQz6qOl/18aibAg+A\nYKUoz4lJTx7L8xW57R56Py390dMoiWULdh/0/3EqKvAACFaG1wrtqWVn8vu9wfMPduqSObetflEA\nsgXnP+v/Y01B4AEQrAQr3Fx6dnrBoZ3eW/mDxeXoGcgWvDOtw8SbJnVO/DzwAAiWzz+sDBtd6Ly3\n1/LTHbsOG79dQhbyW9EXNj27dOWm4B43CJbNq3bPwviO6fb22YsWZPwsLQ/oB2uYwpSkTwoS07Nj\nLRHtyyTmAYI1TEFmwZ2zomx298hDkvMAwVrktW5Fg5fU89dkjovtbuXSmpuKEyogWHt8aufCrXPu\nfYT/tnOMi3MFPcZBAgRrjW97mxOT/vlX+5eDef7Uq0/saTtFq4BgjbEqd5qlw7X9j0YtGKRIfiBY\nS2xPdVu8n4W9FnnXOxHXL1YkSxCsHSpdDGfJjuqdEZubHs/dHPIDo1YBwVrh+JtOb/q+fO72sB0Z\nHOv9TKFsQbBG+LjvraaUDP57a4a7R2TJPsXyBcFa4Myyu7ueeNU+17n2jCM9av4uBbMGwRrgG/eA\nEfbpP3u9GckuJr6FVxQkAoI1QPYKvrZv3H8eCmM57i6F8wbBhKk5zddFn/CdrvgVx669Y7Pi+YNg\notRMGTi637dJ23j+QmTfSa9jKAEEE+SbrvaEdXxl8R9S3q8YnxHKLGd0QDAxzoxjIx3Jrjf52767\nPyNxXIjznFEBwaRY4zA5vHm9cnL5aT9gLAYEk8I7OuGGV9zjOiR+NaCVtz9lA4JJ8HlB++7OyZnJ\ndZ6RpsjxB3AWBYLV58yiiFnrBzkG9kpKslicv+ItDASrzYUH2rXrWLK+2pG4pKPFNBxP2/kyIFhl\n1lrNCebMmh4XR3siI8e2/X6vXECwquyfY503YIvbvGH67k7XqlIiCFaRCg/LsYW51RPCOkfEdj/W\ndgIFAMHqcdBseyKW9eaM+bM7t/AjlQoFwWqxvq/H3I9f5uXyNoQ7/467bdUICFaJGXHLYjj32Zoc\njjXnHFavXBCsDgcSv+dHesIG/jubm4O553slIFgFvnps1drsWn5/FGc1sUvULRsEY2ery+Qa26XT\nmzz/hbPgdvw93ysBwZg5P4y1jr7aPSMxZ9ZdkXeqXz4IxssrJpYzzRrTNb3/8uGjPiQQAAjGyfFb\nWbtnYLZj4nXRYwiFAILxUf98/FWW4jTX701Do7xfEAoCBGNjR1ZqyhpT8VMWD8vaviUVBQjGRPUo\nR1RhfpE3rCTGwd1OLg4QjIcZFktK/OMx85faLVxy60t24wUE4+BMpinak+7suySuS/zIE0RDAcE4\nuDm6+4I/JbaLHZqW8z7hUECw4pS/8HpBp2EfjMgKjw0bdYp0NCBYYc6NTxw435V585DnvFzkJtLR\ngGClOdwv/eCXfSYndeoSz01S7aFvK4BgJamb2Ssj4XX++9+ldUjI2UE6Gj8gWEHWF4944b7hww+e\nLx5OOpRGFBK8NWhFf+MJvtCtXUpWTvEto59+pt17pINpRCHBUcG7CBhN8MzY8L3TdnXvNK5HenKz\n+4CSQbbgMJMAw5kCDxhL8C82qy3C9lzp4tz3ctap/VC/NWQL3tV74t6jR71fHw08YCjB91jNnxV7\nurV/LcE1DufLoOjIr6IvPprztsGr6Pr8NE9S6oLkzLBHXF+SDiYAJe7BPw6Y7DKy4NVZ9uQuSQs7\nPeAw9yL12LdFFGlk1a0srQr60CiC6zaMuePq15JtOVFdOQ2MXAWCYffRM3v93GiMzSnfKc7vPiqr\nelhXNxu+hXQwzYBh99G3b/HTdYj0qHRD/fYBNXd9Oich/9ZC+2rSwTQLjGTJYXPPfpl3fVK6b+Is\n79V7SQfTPLD7qHRO3Owq+uP/Ln56dW5aT3kbK2AEdh+VzAMWZ8zqKTlfjTtXqlm9sPuoZM60Yx1Z\ncdmb+4xIGP4y6WBaAXYflcbdLMPapkYm/754ioYvXx52H5XGhREM53jIFN3VEZnwDOlgWgd2H5XA\n0ZwIe6HDO5eNMncnPumqDWD3UXSGmFgHl5/HJTLsIJyrECoC9INROWlhwlgrG3mVmWXXkA6mbUAw\nIr7WlamTM9zGmpiwk6SDCQEQjMZszuJ0ulNMnN3xN9KxhAQIRmHPCEtkrLtzbLs47mqyb6SEDAhG\nYBoXYbfGWRPNLDOKdCyhAoJDptLB2EdZnfYiG8M+RzqYkAHBobKRYVgrl2UyOdj0WtLBhA4IDpEa\nqz0+ZbzVHW7J0/LQcxAgOES2cV08D3sdJlN30pGgAYJDYe9jjz7vCJ8XNYph9TZPBQSHwMLIHjMG\nJbPOLiZGB2NXVwKC26R2suejc/f9oWsHK+vU6LycVgDBbXEPwzCmRfzgCYcrzpKORQIguA3eYIr7\ndXdZjpUWauF1bnRAcKuczjAzd3ruz2Mfin6SdCzSAMGtsYplTUz4wrgRTNxy0rFIBAS3Qorv9ssx\npvQ+Dstx0rFIBQS3yEUTw/iuYNZn2abqKvyKAoJbYqdPLMNyJtZke5t0LDIAwS0wUfBrYziWYT8l\nHYscQHDzzGYYM+u7Afsu4l9IxyILENwc1akMy5jM/ouYdCwyAcHN8L3DzNpNjNXnV2/PFoIAwcGs\nM3EJFma40Hx+hHQssgHBQXTxVc9xUwS9Fh3N3GgJEBzACV/v18pywzrb2bmkY1ECEHwlxznGY+bC\nOI7lXiAdiyKA4Ct42sxwxT18djm3Ovs3YwcENyVC6Bhxi70M003zb5WFCAi+zAVfw4r1/ePr//Yn\nHYtigOBGfH7Z9ozQemaoaF41AIJFNvqaz0yKMEEnMuhldh0Dgi8xmvHffzP0Pzh5JSC4gSSGMfkV\nMxbSoSgLCPYT5/Nr9/vNIR2KwoBgH/uFlhXr9LWeWW0uOCkDEMzz2xrqZuHZ/iHSsSgOCOY3++67\nXIPic6RjUR4Q3Fd4c8EsGI4gHQoODC842n/tmmwMM5l0KFgwumBzw+ikj+mkQ8GDsQXX+i9fVhC8\nlXQsmDC04AqTTy7L+a5gGuZuNI+RBW/0v5kiXL4c6VDwoYzgmovBn2le8LuCXv/U50jSoWBEtuD/\nlEwpH2C1l+pua7vR/puvcP1mkA4FJ7IFF900P3pu1c+/uzbwgMYFx/p7R5xP8PWkQ8GKbMH2I8eZ\nszx/xBN4QNuCucbhSZoe/jaDbMGxu+qFhcG2dg48oGnBl/z6auldpEPBjGzBC1N38Pz+WbGrAg9o\nWHCdf2RD+M2qz4U3EJAtuH7jPp7fs2xn0AHtCj4gXr4MR71fHJtTXkKzgr9r9Bu0azmFYNiccss8\nP1cNlx4VTnY1+qVr8lULYBjJOvy5n+t+Iy0izDzPiPffNNKhqILRNqcc2+hXmz9/imOwzSnnXnp2\nxDB/JB2KShhrc8rZ/mf7guGHSYeiFobanPIW4cUjzsQybAXpUFTDSJtTWv23X2F++wHSoaiHgTan\nNDf2j34mHYqKGGdzSpv4dN96nnQoamKYGR2RwsikUD2n0fJqd2gYRbBN7P4aYviqCcYQXH/58SDp\nUNTGEILrTZcuX8NdvwYRLLSf/X2kRNKRqI8BBAvXL8uwvlq6mHQoBKBfcHXj9I0s0qGQgHrBFxuH\nN4LG2gwB7YLPN/odTToUMlAu+LLfW0iHQgi6BVc33H6pn93eCnQLdjAc529j9SYdCTFCFFz//Ye7\nEaeYkhdcFye4tfj+HUw6FHKEJrgsL6pbdG4ZUs7kBZv8lbPvVz/SkRAkNMG95p3nz88rQsqZuGAr\nE97Q/+1DOBCihCbYLbwcegRtFRrSgu0Mk+oV6ugbycZBmNAEX/+U77cVE5FyJivY//yIzYw0seUk\nwyBPaIKnm3PH5DIjSksRpJEVzIr935kko9AAoQl+sZHQcyYq2CqujmR0v5T2g69hGC5cmOE+h1wM\nGoFKwQXC/VdYf9Kg489NoVFwuPj+UTypCDQEhYJzGXvDFOg0QgFoCvoElzCMqUhoZI0kU77GoE5w\npn9phn4c8ziR4jUHbYJ/K/Z/g5Z1MiiUCV4qLG4m+A0nULgmoUvwS0LrWej/JqlftkahSvBB8f19\nw72/0DI0Ca7xMFyY/xGw2iVrGIoE1zpZ1srYWYalf3mz0KFHcK2/craYGLZa3YK1DTWC6xpm57AM\nV6VquVqHGsG+zhErjF+Fn1a1WM1Di2Cn/wGw79dPapaqAygRnC/cf4VHDH9XsVBdQIfgIeIApaEn\nUDYLFYJni34zVStSN9Ag+Evf/df/BDis7e8aDgoE/+TvH1nAb7PoX3C10EHyMFRvXyYD/QtunAEN\nA5TNoXvB4gpY7K+qFKc79C44TLx+v1GjNB2ic8Hxol+0V1sNhAKCjwmre14ksjnl3aLfsfjL0imy\nBX/Xic3YwPPlQd9UQfCLot8Y7EXpFtmC+9xduyVpJxHBZaJfO+6SdIxswc6TPP9G4UUSgkW/MEOn\nFWQLzlvL8/Vj5xMQ3LhEMOZy9I1swe+H9zrMHy3IV11wo99KvOXoHPmt6MqXT/F8zct3Bn6OWXCB\n6DeoYKApet199DHRbx7OUigAw+6ja4b4SR0kPao22SL6jcVYCBXocySrDhrQoaLP3UdZ8Bsqutx9\n1CFewLXYiqAGPe4+mir6fQdXCRShw91HF4p+jbtGMAL62320Ep4woKC/3UdFvzAFKyR0t/to4xQs\nLLnTh976wSbRr4H2eJaFzgRPEP3ehiFzKtGX4I2iX6/yeVOKvgTDCCUyuhIMDSx09CTYKvo9q3TO\nFKMjwVNFv7cqnDHV6EfwIdGvSdl8KUc/guERoSR0IxjmUEpDL4KLRL/7lMzVAOhE8DrRbzcFMzUE\nOhEMj5Ckog/B0MCSjC4Eu6GBJRk9CG58C/hhpXI0EHoQLPqFfRgkoAPBcAOWg/YFx8MNWA6aF3y/\n6HedItkZDs0LFv3CRjnS0LpguAHLROOCE+AGLBNtC/5C9LtFfl4GRduCRb/R8rMyKpoWDDdg+WhZ\ncAbcgOWjYcFVot/1MjMyNBoWLFbQMAQtB+0K5uAGrASaFbwYbsCKoFnBot9+snIBtCpYvAHDJCyZ\naFSwHW7ACqFNwfvFCnq/9DwAP9oULFbQNulZAA1oUjD0kJRDi4InQQ9JObQoWPSbJb1wQESDgqGC\nVhLtCY6BClpJtCdY9Hud9KKBy2hOMAxhKYvWBNvgBqwsWhMsVtAbpRcMNEVjgsUKGlbSUQptCbZA\nBa00sgWXiQQekCD4olhBn0JOCrSAbMEljDPJT+ABCYLFCtqMnBJoCflV9PQZzX+OLtgMFbTyyBe8\n6S/Nf44s+BwMYWFAQ40ssYK2SC8TCALD7qM/rPFzzWi0SKAFjQUMu49+udLP4BFokUAFjQXNVNFQ\nQeNBK7uPWqGCxoNWdh+FChoTGtl9FIY4cKGN3UcdUEHjQhu7j4oVNGynojia2H1UnGYHDwmVRwu7\nj6ZCBY0PLfSDxQr6RemFAS2hAcFQQeOEvOBFUEHjhLxgsQucL70ooGWICxaf8sM8aDwQFwxjlHgh\nLRhe9cYMYcFOaGFhhrBgqKBxQ1aw2AWGh0jYICp4KVTQ2CEqWGxhFUsvBWgDkoLFeZTQBcYIScHQ\nwlIBgoLFFhZ0gXFCTvAUaGGpATnBYgtrj/QigLYhJlh8yABPgfFCTDC0sNSBlGCxheWUXgAQCoQE\nF0MLSyUICRZbWLAYB27ICLZAC0styAiGFpZqEBEstrDs0nMHQoSE4DughaUeJASLLaxl0jMHQoWA\nYHFPJHhKqAYEBEMLS03UF2y65NcqPWsgdNQXDC0sVVFdsNjCipGeM4CA2oIT4AJWF7UFs9DCUheV\nBYt7bsAgtFqoLBi6SGqjrmCxiwQTKVVDXcHQwlIdVQWLT5ESpOcKIKKm4AVwAauPmoKhi0QAFQWH\nQxeJACoKhguYBOoJFtd0hzX7VUW9Jf2hhUUE1Zb0F8c44FUGdVFtSX+4gMmg1pL+4hhH0MLwAF5U\nWtL/AlzAhFBpSX+xi9RcawzAiTpL+r8EM2VJgWFzyks0FQxjHMTAsDnlq0P8pA1rkg0MUpIC30jW\nK8sv/52FC5gU+Eaymgp+A6a6kwLfSFZTwfyjPr+jkIMD5INvJOsKwQAp8I1kgWBNgG8kCwRrAnwj\nWSBYE+AbyQLBmkCdfjBADBBMOSCYckAw5eAT/F77wqY4pWCBRDIS2XJ95739QVyCAxhAYaKh5yUk\nGn1SQqLSQxISTfmp7e+A4NYAwU3QuCsQLBeNuwLBctG4KxAsF427AsFyGUphopLgEfi2GXtaQqLr\njkhIdGNF299RTnDQjA9IpIVEME+OckAw5YBgygHBlAOCKQcEUw4IphwQTDlKCd5Z4Jl6DjXRxjxn\nn+/QyyoLa/s7AVSWuHruRk30VJqjuAwxTYmQAPVk+BOhnoyShtDaOhkKCb6Q9syBwUsQE1WGrzmx\nsBNyWRd7I7+lWt/toV9mo45G/2DZ9MuMgUhJNk1nypBPRkMixJPRkCiEk6GQ4E0deP7D9oiJXu7F\n87XsMdSyHp2ILHhHh3q+5mvERJWubaf+NB4pyV9mOMuQT0ZDIsST0ZAohJOhkOBnJ/J8lbUeLdGp\nwzy/JR0xEf9jzl5kwavG3Nx+/M+oqVYwbFQVYpqkMgknQ0iEfDKERCGcDIUEL72J588zyM/J6t9I\nehMxSd3At44iC36QXbHnjt6IicoSPjs3F/WVWOG0I58MvyvUkyEkCuFkKCR45STfD60ZdYmdqnGF\nO5FLup5HF/zXYp4/ZzqKlujBaTxfYzuBlkg47cgnwy8Y9WT4S2r7ZCgkeGNnnv84CzFRTbf5F5FL\nKnVFeZmorWiJNhT7SjMfR0v0wFSer0ZNJJx25JMhJEI+GUKiEE6GUq3oxLWnRy9CTPRyXrkPRMdV\nFRVfcxWIj09r4lZXzS5GS8N/595YNROtFd1w2pFPhpAI+WQIiUI4GYr1g/Mip6I+tL7Tv7AHYsXp\nA72K5rd3Cy8JYfrDlazLcY06gJjGX9uingwhEfLJaLhxq1VFA1oFBFMOCKYcEEw5IJhyQDDlgGDK\nAcGUA4IpBwRTDgimHBBMOSCYckAw5YBgygHBlAOCKQcEUw4IphwQTDlGFXxa2IgvbxvPRwlz3dDf\nZtMNxhX80/HKBan1fNQHx32QDgcfxhXsk3qUOcNHIb9boS+MLLj+yX48CKaU00yE28Z96hMc5na7\nXyIdDj6MK/jr8n0bXLv5qPXl5eVSVpfUCcYVLDSsBj4JVTSlCK3oY5/Y/wWCKUXoBzNxS6GRBegc\nEEw5IJhyQDDlgGDKAcGUA4IpBwRTDgimHBBMOSCYckAw5YBgygHBlAOCKQcEUw4IphwQTDkgmHL+\nH1m/UyfleiRPAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "tempData <- data.frame(data$BF, data$pval.TADA)\n",
    "\n",
    "tempData <- tempData[tempData[, 1] < quantile(tempData[, 1], 0.99), ]\n",
    "\n",
    "plot(tempData[, 1], tempData[, 2], xlab = \"BF\", ylab = \"p\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Extract genes whose FDRs < 0.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32\n",
      "ADNP ANK2 APH1A ARID1B ASH1L ASXL3 BCL11A BIRC6 CACNA2D3 CHD8 CUL3 DPP3 DYRK1A GABRB3 GALNTL4 GRIN2B KATNAL2 KIAA0182 MLL3 MYO9B NAA15 NR3C2 POGZ PTEN PYHIN1 RELN SCN2A SUV420H1 SYNGAP1 TBR1 TGM1 TRIO "
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat data/ext_TADA_results.csv |awk -F\",\" '$(NF-1)<0.1'|\\\n",
    "    awk -F\",\" '{print $1}'|sed 's/\"//g'|grep -v \"Gene\" |sort > data/list.genesWithNewPiFDR01.txt\n",
    "    \n",
    "cat data/list.genesWithNewPiFDR01.txt|wc -l\n",
    "cat data/list.genesWithNewPiFDR01.txt|tr \"\\n\" \" \""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Intersect this gene set with the 33 genes in AUT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "join -1 1 -2 1 data/list.genesWithNewPiFDR01.txt data/list.33genes.FDR0.1.txt|wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "####Test with known risk genes from three diseases: ASD + EPI + ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               [,1]     [,2]\n",
       "gamma.mean.dn  14.0    2.330\n",
       "beta.dn         4.8    2.325\n",
       "gamma.mean.CC   2.3    1.000\n",
       "beta.CC         4.0 1000.000\n",
       "rho1            0.1    0.500\n",
       "nu1           200.0  100.000\n",
       "rho0            0.1    0.500\n",
       "nu0           200.0  100.000\n",
       "               [,1]     [,2]\n",
       "gamma.mean.dn  14.0    2.330\n",
       "beta.dn         4.8    2.325\n",
       "gamma.mean.CC   2.3    1.000\n",
       "beta.CC         4.0 1000.000\n",
       "rho1            0.1    0.500\n",
       "nu1           200.0  100.000\n",
       "rho0            0.1    0.500\n",
       "nu0           200.0  100.000\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "hyperpar1 <- hyperpar\n",
    "hyperpar1[3, 1] <- 2.3\n",
    "hyperpar1\n",
    "data1 <- data\n",
    "head(data1$mut.rate)\n",
    "print(hyperpar)\n",
    "print(hyperpar1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "#Running TADA\n",
    "rs1 <- TADA(counts, N, data1$mut.rate, mu.frac, hyperpar1)\n",
    "data1$BF <- rs1$BF.total\n",
    "# Estimating p-values of BFs (this is optional and slow)\n",
    "#rsp1 <- TADAp(counts, N, data1$mut.rate, mu.frac, hyperpar1, l=100)\n",
    "#data1$pval.TADA <- rsp1$pval\n",
    "\n",
    "# FDR estimation\n",
    "data1 <- data1[order(-data1$BF),]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       Gene mut.rate dn.LoF case.LoF ctrl.LoF trans.LoF ntrans.LoF dn.mis3\n",
       "15631 STAT4 2.61e-05      0        0        1         0          1       0\n",
       "      case.mis3 ctrl.mis3 trans.mis3 ntrans.mis3     BF.dn          BF\n",
       "15631         3        15          1           5 0.7426887 2.21841e+18\n",
       "      qvalue.dn    qvalue  pval.TADA\n",
       "15631 0.9324757 0.9257113 0.02915459\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "head(data1, 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       Gene mut.rate dn.LoF case.LoF ctrl.LoF trans.LoF ntrans.LoF dn.mis3\n",
       "14230 SCN2A  7.4e-05      9        2        0         0          0      12\n",
       "      case.mis3 ctrl.mis3 trans.mis3 ntrans.mis3     BF.dn           BF\n",
       "14230         4        26          5           5 615476690 4.745777e+17\n",
       "      qvalue.dn qvalue pval.TADA\n",
       "14230  2.55e-08      0         1\n",
       "       Gene mut.rate dn.LoF case.LoF ctrl.LoF trans.LoF ntrans.LoF dn.mis3\n",
       "15631 STAT4 2.61e-05      0        0        1         0          1       0\n",
       "      case.mis3 ctrl.mis3 trans.mis3 ntrans.mis3     BF.dn        BF qvalue.dn\n",
       "15631         3        15          1           5 0.7426887 0.3747428 0.9324757\n",
       "         qvalue  pval.TADA\n",
       "15631 0.9257113 0.02915459\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "print(head(data, 1))\n",
    "data[grep(\"STAT4\", data$Gene), ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        Gene mut.rate dn.LoF case.LoF ctrl.LoF trans.LoF ntrans.LoF dn.mis3\n",
       "15631  STAT4 2.61e-05      0        0        1         0          1       0\n",
       "8972  LINGO1 3.98e-05      0        0        0         0          0       0\n",
       "8220  KCNK12 1.41e-05      0        0        0         0          0       1\n",
       "5108    ELL3 1.48e-05      0        3        8         1          1       0\n",
       "6364  GARNL3 4.02e-05      0        2        1         0          0       0\n",
       "11342 OR52B2 1.25e-05      0        2        3         3          1       0\n",
       "      case.mis3 ctrl.mis3 trans.mis3 ntrans.mis3     BF.dn           BF\n",
       "15631         3        15          1           5 0.7426887 2.218410e+18\n",
       "8972          0         9          1           1 0.6370376 1.116470e+10\n",
       "8220          0         0          0           0 0.8508121 5.407493e+09\n",
       "5108          0         6          4           1 0.8439596 3.030332e+08\n",
       "6364          6        16          6           7 0.6342135 1.731686e+07\n",
       "11342         0         6          0           3 0.8667740 9.846528e+06\n",
       "      qvalue.dn       qvalue  pval.TADA\n",
       "15631 0.9324757 0.000000e+00 0.02915459\n",
       "8972  0.9367594 5.588556e-10 0.02419482\n",
       "8220  0.9178345 1.141806e-09 0.07495849\n",
       "5108  0.9193825 1.115135e-08 0.02916401\n",
       "6364  0.9368288 1.530453e-07 0.07506373\n",
       "11342 0.9129570 3.387609e-07 0.16463245\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "data1$qvalue <- Bayesian.FDR(data1$BF, pi0)$FDR\n",
    "\n",
    "head(data1[data1$qvalue < 0.1, ])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] 1\n",
       "[1] 0.9991408\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "print(cor(data$BF, data1$BF))\n",
    "print(cor(data$qvalue, data1$qvalue))\n",
    "#print(head(data, 3))\n",
    "#print(head(data1, 3))\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1] \"/usr/data/extTADA\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "write.csv(data1, \"data/ext_TADA_results_usingThreeGeneSets.csv\", row.names=FALSE)\n",
    "\n",
    "getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "104\n",
      "ALAS1 ANKRD39 ANKRD49 ARL11 ARMC3 BHMT2 C12orf29 C22orf24 C5orf34 CAPRIN2 CASP1 CASP2 CDR2 CH25H CHN2 CHRNB1 CLK2 CLPS COX6B2 CSF1R CTDSP1 CYP4A11 DGKE DIS3L DSCR4 ELL3 EML1 EN1 ERV3-1 ESM1 FAM108A1 FAM189A2 FOXJ1 GARNL3 GPI GPR123 GRM4 HDAC9 HPSE2 ISM1 KCNK12 KCNMB4 KIF4B KRT14 KTN1 LINGO1 LRIT2 MED26 MIR205HG MRGPRF MRM1 MT1E MTX1 NPRL2 OPN1MW OR2M4 OR2Y1 OR52B2 OR6K2 OR8S1 PAN3 PCDHB12 PDPK1 PEAK1 PHGDH PIAS2 PPP2CB PRAMEF1 PRAP1 PTPRCAP PXDC1 RECK RHOH RIOK1 RLN1 SAGE1 SAPCD2 SCGB1A1 SETD8 SH3YL1 SLC29A4 SNAI2 SRCAP SRD5A2 ST6GALNAC5 STARD9 STAT4 STK11IP TAS2R9 TBCK TFDP1 THSD1 TIMM8B TP53I11 TSN UQCR11 VPS53 VPS54 XAGE3 ZBTB16 ZNF148 ZNF177 ZNF664-FAM101A ZNF706 "
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat data/ext_TADA_results_usingThreeGeneSets.csv |awk -F\",\" '$(NF-1)<0.1'|\\\n",
    "    awk -F\",\" '{print $1}'|sed 's/\"//g'|grep -v \"Gene\" |\\\n",
    "    sort > data/list.genesWithNewPiFDR01.usingThreeGeneSets.txt\n",
    "    \n",
    "cat data/list.genesWithNewPiFDR01.usingThreeGeneSets.txt|wc -l\n",
    "cat data/list.genesWithNewPiFDR01.usingThreeGeneSets.txt|tr \"\\n\" \" \"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "join -1 1 -2 1 data/list.genesWithNewPiFDR01.usingThreeGeneSets.txt data/list.33genes.FDR0.1.txt|wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#!gist -p Re_run_TADA_exTADA.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://gist.github.com/40eb110d8ac72b91d6e9\r\n"
     ]
    }
   ],
   "source": [
    "!gist -u https://gist.github.com/40eb110d8ac72b91d6e9 Re_run_TADA_exTADA.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
