{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Some main steps:\n",
    "\n",
    " Download dbNSFP1.3_a \n",
    " \n",
    " Use pseq to annotation all positions inside this database\n",
    " \n",
    " Convert to VCF\n",
    " \n",
    " Extract variants\n",
    "\n",
    "\n",
    "Notes:\n",
    "In dbNSFP, there are positions whose ref and alt are the same, \n",
    "\n",
    "$=>$ pseq: we cannot get \"snp_fw_seq3\" (REF = ALT)\n",
    "\n",
    "Ex:\n",
    "\n",
    "chrM:8531\tallele_ref\tA\n",
    "\n",
    "chrM:8531\tallele_alt\tA\n",
    "\n",
    "chrM:8531\tseqdb_ref\tA\n",
    "\n",
    "chrM:8531\tfunc_alt\t.\n",
    "\n",
    "chrM:8531\tfunc\tintergenic-region\n",
    "\n",
    "chrM:8531\ttranscript\t.\n",
    "\n",
    "chrM:8531\ttranscript_strand\t.\n",
    "\n",
    "\n",
    "Some small differences between Old and New data sets\n",
    "\n",
    "result_mutation_gene_damaging.7methods.txt: use old gene names\n",
    "\n",
    "Result_7methods_allGenes_trinucleotide.txt.join.192_mutationRate_forGenes: new gene names\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Convert to an input file for pseq\n",
    "\n",
    "zcat db.shortened.dbnsfp31a.gz|awk 'NR>1 {print $1\":\"$2,$5,$6,$7}' > all_coordinate_hg19_dbnsfp31a.txt &\n",
    "\n",
    "## Add chromosomes\n",
    "\n",
    "awk -F\":\" '{print $1, $0}' all_coordinate_hg19_dbnsfp31a.txt > all_coordinate_hg19_dbnsfp31a.txt_singleChr.txt\n",
    "\n",
    "## Split into chromosomes\n",
    "\n",
    "In R\n",
    "\n",
    "library(\"data.table\")\n",
    "aa <- fread(\"all_coordinate_hg19_dbnsfp31a.txt_singleChr.txt\", header = FALSE)\n",
    "\n",
    "setkey(aa, V1)\n",
    "\n",
    "x2 <- table(x1)\n",
    "x2 <- names(x2)\n",
    "\n",
    "for (ii in (1:length(x2))) {\n",
    " xTemp <- aa[x2[ii]]\n",
    " write.table(xTemp, paste(\"SingleChr_annotation/All_coordinate_chr\", x2[ii], \".txt\", sep = \"\"), col.names=FALSE, row.names=FALSE, quote=FALSE) }\n",
    "\n",
    "## Extract trinucleotide context (fast)\n",
    "\n",
    "for file in $(ls Reduced.All_coordinate_chr*annotated|grep -v \"chr2\\.\")\n",
    "\n",
    "do ./convert2columns.sh $file\n",
    "\n",
    "done \n",
    "\n",
    "\n",
    "## Create vcf files (fast)\n",
    "\n",
    "################\n",
    "Use the file \n",
    "\n",
    "convert2VCF_and_filter.sh\n",
    "\n",
    "## Combine all files\n",
    "\n",
    "outFile=Result_7methods_allGenes_trinucleotide.txt\n",
    "\n",
    "rm -f ${outFile}\n",
    "\n",
    "for file in $(ls Extract.damaging.7methods.Reduced.All_coordinate_chr*)\n",
    "do\n",
    "\n",
    "grep -v \"#\" $file|awk -F\";\" '{print $1}'|awk 'NF>7'|\\\n",
    "    awk '{print $(NF-1),$NF}'|sort|\\\n",
    "    uniq -c|awk '{print $3,$2,$1}' >> ${outFile}\n",
    "    \n",
    "done\n",
    "\n",
    "\n",
    "## Join 192\n",
    "\n",
    "cat Result_7methods_allGenes_trinucleotide.txt |sort|join -1 1 -2 1 - ~/psychen/resources/genic_mutation_w_annotation/192_trinuc_mutations.sort.txt > Result_7methods_allGenes_trinucleotide.txt.join.192\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
